src/vault/authenticate_vault/authenticate_vault_trait.rs

```diff
@@ -0,0 +1,30 @@
+use crate::errors::error::VaultError;
+use crate::vault::vault_config::VaultConfig;
+use crate::vault::vault_service::VaultService;
+use vaultrs::{
+    client::{VaultClient, VaultClientSettingsBuilder},
+};
+use async_trait::async_trait;
+
+#[async_trait]
+pub trait AuthenticateVault {
+    async fn authenticate(&self, config: VaultConfig) -> Result<VaultService, VaultError> {
+        let settings: VaultClientSettingsBuilder = self.get_vault_settings(config.clone());
+
+        let jwt_token = self.get_jwt_token(config.clone()).unwrap();
+
+        let client: VaultClient = self.create_client(settings);
+
+        return self.create_service(client, config, jwt_token).await;
+    }
+
+    fn get_vault_settings(&self, config: VaultConfig) -> VaultClientSettingsBuilder;
+
+    fn get_jwt_token(&self, config: VaultConfig) -> Result<Option<String>, VaultError>;
+
+    fn create_client(&self, settings: VaultClientSettingsBuilder) -> VaultClient {
+        return VaultClient::new(settings.build().unwrap()).unwrap();
```

Comment:
I've seen a lot of posts about the unwrap method, most of them say it's a bad practice to use, and some of them say that it can be used.

What I understand from what I've read is that it's better to use pattern-matching instead of unwrap here, but I'm not expert on this topic. What is the standard Valensas follows?